src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -507,6 +524,26 @@ public object GetOrRegisterObjectForComInstance(IntPtr externalComObject, Create
                         retValue = handle.Target;
                         return true;
                     }
+
+                    if (wrapperMaybe is not null)
+                    {
+                        retValue = wrapperMaybe;
+                        NativeObjectWrapper wrapper = new NativeObjectWrapper(
+                            externalComObject,
+                            this,
+                            retValue);
+                        if (!_rcwTable.TryGetValue(retValue, out var existingWrapper))
+                        {
+                            _rcwTable.Add(retValue, wrapper);
```

Comment:
Can this just call `TryAdd`?

Reply:
Both `Dictionary` and `ConditionalWeakTable` does not have this method. And because this is under lock I think this should be safe

Reply:
`_rcwTable` is `ConditionalWeakTable`. `ConditionalWeakTable` has `TryAdd`: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2.tryadd?view=net-7.0

Reply:
Now I feel ashamed, because I look at it, but VS witout project pickup Net Framework version... Should be done

Reply:
> Now I feel ashamed, because I look at it, but VS witout project pickup Net Framework version... Should be done

You can edit these with the src\coreclr\nativeaot\nativeaot.sln solution file. Might need to build the repo first and then choose the debug/release x64/whatever configuration in VS that matches how you built the repo, but it works well, including intellisense.

Reply:
Yeah. I know. I just start debugging native Exe, stop in the debugger and continue editing from that point. That was convinient but definitely not optimal experience which led me to this place.